design/pipeline-resources.md

```diff
@@ -228,17 +228,16 @@ other repositories' checkout directory: `$PIPELINES_RESOURCESDIR/<repository-ide
 
 If you need to consume a container image as part of your CI/CD pipeline, you can achieve it using `containers`. A container can be an Azure Container Registry or any external Docker registry.
 
-### Schema
 
+If you need to consume images from external docker registries as part of your pipeline, you can define a generic container resource. A generic container resource requires a Docker registry service connection.
+
+### Schema
 ```yaml
 resources:          # types: pipelines | repositories | containers | packages
   containers:
-  - container: string # identifier for the container resource      
-    type: enum # type of the registry like ACR, Docker etc. 
-    connection: string # service connection to connect to the image registry, defaults to ACR??
+  - container: string # identifier for the container resource   
+    connection: string # service connection (Docker registry) to connect to the image registry;
     image: string # container image name, Tag/Digest is optional; defaults to latest image
-    options: string # arguments to pass to container at startup
```

Comment:
You can't remove `options`, `env`, `ports`, and `volumes`. They're used in container scenarios today. See https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema#resources

Reply:
Thanks for the feedback Matt. Yes, it should support other container scenarios as you mentioned and we will continue supporting that. Here I have focused on container CD scenarios. I will mention about that.

Reply:
Updated the schema as suggested